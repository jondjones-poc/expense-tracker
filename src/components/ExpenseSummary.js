import numeral from 'numeral';
import React from 'react'
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { adminPageCreateLink } from '../actions/urls';

import getExpenseTotal from '../selectors/expense-total';
import selectExpenses from '../selectors/expenses';

export const ExpenseSummary = ({totalItems, totalAmount}) => {
        const formattedTotalAmount = numeral(totalAmount / 100).format('$0,0.00');
        return (
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">
                        <span>You have {totalItems}</span> expense(s) totaling <span>{formattedTotalAmount}</span>
                    </h1>
                    <div className="page-header__actions">
                        <Link to={adminPageCreateLink} className="button" >Create An Expense</Link>
                    </div>
                </div>
            </div>
    )};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    const totalAmount = getExpenseTotal(visibleExpenses);

    return {
        totalAmount: totalAmount,
        totalItems: visibleExpenses.length
    }
};

export default connect(mapStateToProps)(ExpenseSummary);