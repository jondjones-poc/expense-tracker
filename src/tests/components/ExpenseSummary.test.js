import React from 'react';
import { shallow } from 'enzyme';
import ExpenseSummary from '../../components/expenseListItem';
import expensesTestData from '../Fixtures/expensesTestdata';

test('Should render expense list item correctly', () => {
    const wrapper = shallow(<ExpenseSummary />);
    expect(wrapper).toMatchSnapshot();
});
