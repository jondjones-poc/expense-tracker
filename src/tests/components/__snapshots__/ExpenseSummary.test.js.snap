// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense list item correctly 1`] = `
<li>
  <Link
    replace={false}
    to="/edit/undefined"
  >
    <h3 />
  </Link>
  <p>
    $0.00
    -
    January 1st, 1970
  </p>
</li>
`;
